## APPLIED STATISTICS

### Turma 10DTSR

### Trabalho prático usando o Python

## Habilidades desenvolvidas: análise descritiva dos dados.



### Instalar as bibliotecas pandas, numpy, matplotlib e seaborn

In [79]:
# Importar as bibliotecas
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns



import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)


In [80]:
# Importar a base de dados
cadastro = pd.read_csv('cadastro.csv')

In [81]:
# Visualizar os dados
cadastro.head().T

0                 1  \
Unnamed: 0                                     1                 2   
NUM_CPF                              40977545806       72544350725   
CHAVE_CONTRATO                                 1                 2   
DATA_NASCIMENTO                       25/09/1996        10/03/1961   
RENDA_PRESUMIDA                       1890000.00        1333890.24   
CEP                                      2471020           1319001   
LOGRADOURO            R AUGUSTO VIANA DE CASTELO     R MARIA PAULA   
NUMERO                                     26.00            201.00   
COMPLEMENTO                                  NaN             AP 52   
CEP_A                                    2471020           1319001   
BAIRRO                             VILA BASILEIA        BELA VISTA   
CIDADE                                 SAO PAULO         SAO PAULO   
UF                                            SP                SP   
LATITUDE                                     NaN               NaN   
LONGITUDE                                    NaN               NaN   
DDD_CELULAR                                11.00             11.00   
CELULAR                             977060299.00      972545525.00   
DDD_CELULAR_2                                NaN             11.00   
CELULAR_2                                    NaN      985714369.00   
COD_BANCO                                    NaN            341.00   
NUM_AGENCIA                                  NaN           6571.00   
NUM_CONTA                                    NaN          15044.00   
MARCA_VEICULO                                NaN           FERRARI   
MODELO_VEICULO                               NaN        355 SPIDER   
PLACA_VEICULO                                NaN           AFZ6118   
CNPJ_CREDOR                     3774819001508.00 40432544000147.00   
VALOR_DIVIDA                                 220               850   
STATUS_CONSENTIMENTO                           1                 1   
DATA_INCLUSAO1                2/26/2019 11:28:49  2/7/2001 0:00:00   
IDADE                                      23.00             40.00   

                                      2                3  \
Unnamed: 0                            3                4   
NUM_CPF                     10536099812       1471742873   
CHAVE_CONTRATO                        3                4   
DATA_NASCIMENTO              28/11/1965       02/05/1959   
RENDA_PRESUMIDA                22000.00         16000.00   
CEP                             4323120          4136030   
LOGRADOURO                  R DAS JOIAS      R GUARAREMA   
NUMERO                           202.00           430.00   
COMPLEMENTO                       VL DO              NaN   
CEP_A                           4323120          4136030   
BAIRRO                 VILA DO ENCONTRO  BOSQUE DA SAUDE   
CIDADE                        SAO PAULO        SAO PAULO   
UF                                   SP               SP   
LATITUDE                         -23.66           -23.61   
LONGITUDE                        -46.64           -46.63   
DDD_CELULAR                       11.00              NaN   
CELULAR                    991131317.00              NaN   
DDD_CELULAR_2                     11.00              NaN   
CELULAR_2                  998095934.00              NaN   
COD_BANCO                        748.00           104.00   
NUM_AGENCIA                     7006.00          1850.00   
NUM_CONTA                      52946.00       1103486.00   
MARCA_VEICULO                       NaN              NaN   
MODELO_VEICULO                      NaN              NaN   
PLACA_VEICULO                       NaN              NaN   
CNPJ_CREDOR           61695227000193.00              NaN   
VALOR_DIVIDA                        142                0   
STATUS_CONSENTIMENTO                  1                1   
DATA_INCLUSAO1        8/23/2018 6:42:12              NaN   
IDADE                             53.00              NaN   

               

In [82]:
# Tipos de variáveis
cadastro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 426 entries, 0 to 425
Data columns (total 30 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            426 non-null    int64  
 1   NUM_CPF               426 non-null    int64  
 2   CHAVE_CONTRATO        426 non-null    int64  
 3   DATA_NASCIMENTO       426 non-null    object 
 4   RENDA_PRESUMIDA       426 non-null    float64
 5   CEP                   426 non-null    int64  
 6   LOGRADOURO            426 non-null    object 
 7   NUMERO                425 non-null    float64
 8   COMPLEMENTO           206 non-null    object 
 9   CEP_A                 426 non-null    int64  
 10  BAIRRO                390 non-null    object 
 11  CIDADE                426 non-null    object 
 12  UF                    426 non-null    object 
 13  LATITUDE              389 non-null    float64
 14  LONGITUDE             389 non-null    float64
 15  DDD_CELULAR           2

In [83]:
# Contagem dos dados
cadastro.count()

Unnamed: 0              426
NUM_CPF                 426
CHAVE_CONTRATO          426
DATA_NASCIMENTO         426
RENDA_PRESUMIDA         426
CEP                     426
LOGRADOURO              426
NUMERO                  425
COMPLEMENTO             206
CEP_A                   426
BAIRRO                  390
CIDADE                  426
UF                      426
LATITUDE                389
LONGITUDE               389
DDD_CELULAR             275
CELULAR                 275
DDD_CELULAR_2           170
CELULAR_2               170
COD_BANCO               308
NUM_AGENCIA             308
NUM_CONTA               308
MARCA_VEICULO             2
MODELO_VEICULO            2
PLACA_VEICULO             2
CNPJ_CREDOR             421
VALOR_DIVIDA            426
STATUS_CONSENTIMENTO    426
DATA_INCLUSAO1          421
IDADE                   421
dtype: int64

In [84]:
# Contagem de valores únicos
cadastro.nunique()

Unnamed: 0              426
NUM_CPF                 311
CHAVE_CONTRATO          426
DATA_NASCIMENTO         305
RENDA_PRESUMIDA         317
CEP                     310
LOGRADOURO              308
NUMERO                  241
COMPLEMENTO              86
CEP_A                   310
BAIRRO                  230
CIDADE                    1
UF                        1
LATITUDE                282
LONGITUDE               282
DDD_CELULAR              18
CELULAR                 202
DDD_CELULAR_2            14
CELULAR_2               124
COD_BANCO                17
NUM_AGENCIA             294
NUM_CONTA               308
MARCA_VEICULO             2
MODELO_VEICULO            2
PLACA_VEICULO             2
CNPJ_CREDOR             153
VALOR_DIVIDA            235
STATUS_CONSENTIMENTO      1
DATA_INCLUSAO1          405
IDADE                    54
dtype: int64

In [85]:
# Retorna verdadeiro ou falso para presença de dados faltantes (missing value)
cadastro.isna()

Unnamed: 0  NUM_CPF  CHAVE_CONTRATO  DATA_NASCIMENTO  RENDA_PRESUMIDA  \
0         False    False           False            False            False   
1         False    False           False            False            False   
2         False    False           False            False            False   
3         False    False           False            False            False   
4         False    False           False            False            False   
..          ...      ...             ...              ...              ...   
421       False    False           False            False            False   
422       False    False           False            False            False   
423       False    False           False            False            False   
424       False    False           False            False            False   
425       False    False           False            False            False   

       CEP  LOGRADOURO  NUMERO  COMPLEMENTO  CEP_A  ...  NUM_AGENCIA  \
0    False       False   False         True  False  ...         True   
1    False       False   False        False  False  ...        False   
2    False       False   False        False  False  ...        False   
3    False       False   False         True  False  ...        False   
4    False       False   False        False  False  ...        False   
..     ...         ...     ...          ...    ...  ...          ...   
421  False       False   False         True  False  ...        False   
422  False       False   False        False  False  ...         True   
423  False       False   False         True  False  ...         True   
424  False       False   False        False  False  ...        False   
425  False       False   False        False  False  ...        False   

     NUM_CONTA  MARCA_VEICULO  MODELO_VEICULO  PLACA_VEICULO  CNPJ_CREDOR  \
0         True           True            True           True        False   
1        False          False           False          False        False   
2        False           True            True           True        False   
3        False           True            True           True         True   
4        False           True            True           True        False   
..         ...            ...             ...            ...          ...   
421      False           True            True           True        False   
422       True           True            True           True        False   
423       True           True            True           True        False   
424      False           True            True           True        False   
425      False           True            True           True        False   

     VALOR_DIVIDA  STATUS_CONSENTIMENTO  DATA_INCLUSAO1  IDADE  
0           False                 False           False  False  
1           False                 False           False  False  
2           False                 False           False  False  
3           False                 False            True   True  
4           False                 False           False  False  
..            ...                   ...             ...    ...  
421         False                 False           False  False  
422         False                 False           False  False  
423         False                 False           False  False  
424         False                 False           False  False  
425         False                 False           False  False  

[426 rows x 30 columns]

In [86]:
# Soma o número de dados faltantes (missing value)
cadastro.isna().sum()

Unnamed: 0                0
NUM_CPF                   0
CHAVE_CONTRATO            0
DATA_NASCIMENTO           0
RENDA_PRESUMIDA           0
CEP                       0
LOGRADOURO                0
NUMERO                    1
COMPLEMENTO             220
CEP_A                     0
BAIRRO                   36
CIDADE                    0
UF                        0
LATITUDE                 37
LONGITUDE                37
DDD_CELULAR             151
CELULAR                 151
DDD_CELULAR_2           256
CELULAR_2               256
COD_BANCO               118
NUM_AGENCIA             118
NUM_CONTA               118
MARCA_VEICULO           424
MODELO_VEICULO          424
PLACA_VEICULO           424
CNPJ_CREDOR               5
VALOR_DIVIDA              0
STATUS_CONSENTIMENTO      0
DATA_INCLUSAO1            5
IDADE                     5
dtype: int64

In [87]:
# Calcular o porcentagem de dados faltantes (missing value)
cadastro.isna().sum()/len(cadastro)*100

Unnamed: 0              0.00
NUM_CPF                 0.00
CHAVE_CONTRATO          0.00
DATA_NASCIMENTO         0.00
RENDA_PRESUMIDA         0.00
CEP                     0.00
LOGRADOURO              0.00
NUMERO                  0.23
COMPLEMENTO            51.64
CEP_A                   0.00
BAIRRO                  8.45
CIDADE                  0.00
UF                      0.00
LATITUDE                8.69
LONGITUDE               8.69
DDD_CELULAR            35.45
CELULAR                35.45
DDD_CELULAR_2          60.09
CELULAR_2              60.09
COD_BANCO              27.70
NUM_AGENCIA            27.70
NUM_CONTA              27.70
MARCA_VEICULO          99.53
MODELO_VEICULO         99.53
PLACA_VEICULO          99.53
CNPJ_CREDOR             1.17
VALOR_DIVIDA            0.00
STATUS_CONSENTIMENTO    0.00
DATA_INCLUSAO1          1.17
IDADE                   1.17
dtype: float64

In [88]:
# Análise descritiva das variáveis
cadastro.describe()


Unnamed: 0        NUM_CPF  CHAVE_CONTRATO  RENDA_PRESUMIDA        CEP  \
count      426.00         426.00          426.00           426.00     426.00   
mean       213.50 26830502524.11          213.50         10387.81 4635212.33   
std        123.12 19135846183.67          123.12        111783.91 1978839.32   
min          1.00    94700508.00            1.00             0.00 1007020.00   
25%        107.25 10869067862.00          107.25          2068.50 3153250.25   
50%        213.50 26041138816.00          213.50          2431.58 4323120.00   
75%        319.75 35798933865.00          319.75          2923.55 5644782.50   
max        426.00 99921022504.00          426.00       1890000.00 8490018.00   

        NUMERO      CEP_A  LATITUDE  LONGITUDE  DDD_CELULAR      CELULAR  \
count   425.00     426.00    389.00     389.00       275.00       275.00   
mean    595.29 4635212.33    -23.58     -46.60        16.50 976736587.11   
std    4869.60 1978839.32      0.08       0.11        17.78  56323647.16   
min       3.00 1007020.00    -23.80     -46.81        11.00  78462120.00   
25%      58.00 3153250.25    -23.63     -46.68        11.00 969555131.50   
50%     164.00 4323120.00    -23.56     -46.63        11.00 982497429.00   
75%     430.00 5644782.50    -23.51     -46.52        11.00 992424488.50   
max   99999.00 8490018.00    -23.43     -46.38        98.00 999993969.00   

       DDD_CELULAR_2    CELULAR_2  COD_BANCO  NUM_AGENCIA     NUM_CONTA  \
count         170.00       170.00     308.00       308.00        308.00   
mean           18.26 960391066.33     189.76      2570.55   51862517.85   
std            19.58 138162785.25     175.78      2347.82  469016468.99   
min            11.00  77151941.00       1.00         1.00        290.00   
25%            11.00 973504991.25      33.00       563.25      15030.00   
50%            11.00 982267694.50     237.00      1900.00      53951.00   
75%            11.00 993495586.00     341.00      3781.00     488489.25   
max            98.00 999889639.00     756.00      9270.00 6500059293.00   

            CNPJ_CREDOR  VALOR_DIVIDA  STATUS_CONSENTIMENTO   IDADE  
count            421.00        426.00                426.00  421.00  
mean  31873758302107.55     212868.56                  1.00   45.74  
std   22283968587101.65    2202077.89                  0.00  107.73  
min     113451000135.00          0.00                  1.00   17.00  
25%    8623036000133.00        198.50                  1.00   30.00  
50%   40432544000147.00        360.00                  1.00   37.00  
75%   43374768000138.00        981.00                  1.00   46.00  
max   96367305000105.00   28072009.00                  1.00 2006.00

In [89]:
# Análise descritiva das variáveis
cadastro.describe().round(2)


Unnamed: 0        NUM_CPF  CHAVE_CONTRATO  RENDA_PRESUMIDA        CEP  \
count      426.00         426.00          426.00           426.00     426.00   
mean       213.50 26830502524.11          213.50         10387.81 4635212.33   
std        123.12 19135846183.67          123.12        111783.91 1978839.32   
min          1.00    94700508.00            1.00             0.00 1007020.00   
25%        107.25 10869067862.00          107.25          2068.50 3153250.25   
50%        213.50 26041138816.00          213.50          2431.58 4323120.00   
75%        319.75 35798933865.00          319.75          2923.54 5644782.50   
max        426.00 99921022504.00          426.00       1890000.00 8490018.00   

        NUMERO      CEP_A  LATITUDE  LONGITUDE  DDD_CELULAR      CELULAR  \
count   425.00     426.00    389.00     389.00       275.00       275.00   
mean    595.29 4635212.33    -23.58     -46.60        16.50 976736587.11   
std    4869.60 1978839.32      0.08       0.11        17.78  56323647.16   
min       3.00 1007020.00    -23.80     -46.81        11.00  78462120.00   
25%      58.00 3153250.25    -23.63     -46.68        11.00 969555131.50   
50%     164.00 4323120.00    -23.56     -46.63        11.00 982497429.00   
75%     430.00 5644782.50    -23.51     -46.52        11.00 992424488.50   
max   99999.00 8490018.00    -23.43     -46.38        98.00 999993969.00   

       DDD_CELULAR_2    CELULAR_2  COD_BANCO  NUM_AGENCIA     NUM_CONTA  \
count         170.00       170.00     308.00       308.00        308.00   
mean           18.26 960391066.33     189.76      2570.55   51862517.85   
std            19.58 138162785.25     175.78      2347.82  469016468.99   
min            11.00  77151941.00       1.00         1.00        290.00   
25%            11.00 973504991.25      33.00       563.25      15030.00   
50%            11.00 982267694.50     237.00      1900.00      53951.00   
75%            11.00 993495586.00     341.00      3781.00     488489.25   
max            98.00 999889639.00     756.00      9270.00 6500059293.00   

            CNPJ_CREDOR  VALOR_DIVIDA  STATUS_CONSENTIMENTO   IDADE  
count            421.00        426.00                426.00  421.00  
mean  31873758302107.56     212868.56                  1.00   45.74  
std   22283968587101.65    2202077.89                  0.00  107.73  
min     113451000135.00          0.00                  1.00   17.00  
25%    8623036000133.00        198.50                  1.00   30.00  
50%   40432544000147.00        360.00                  1.00   37.00  
75%   43374768000138.00        981.00                  1.00   46.00  
max   96367305000105.00   28072009.00                  1.00 2006.00

In [90]:
# Mudar o formato da saída da análise descritiva
Describe=cadastro.describe().round(2)
print(Describe)

       Unnamed: 0        NUM_CPF  CHAVE_CONTRATO  RENDA_PRESUMIDA        CEP  \
count      426.00         426.00          426.00           426.00     426.00   
mean       213.50 26830502524.11          213.50         10387.81 4635212.33   
std        123.12 19135846183.67          123.12        111783.91 1978839.32   
min          1.00    94700508.00            1.00             0.00 1007020.00   
25%        107.25 10869067862.00          107.25          2068.50 3153250.25   
50%        213.50 26041138816.00          213.50          2431.58 4323120.00   
75%        319.75 35798933865.00          319.75          2923.54 5644782.50   
max        426.00 99921022504.00          426.00       1890000.00 8490018.00   

        NUMERO      CEP_A  LATITUDE  LONGITUDE  DDD_CELULAR      CELULAR  \
count   425.00     426.00    389.00     389.00       275.00       275.00   
mean    595.29 4635212.33    -23.58     -46.60        16.50 976736587.11   
std    4869.60 1978839.32      0.08       0.11     

In [91]:
pd.set_option('display.float_format','{:.2f}'.format)

In [92]:
# Análise descritiva das variáveis
cadastro.describe().T


count              mean               std  \
Unnamed: 0           426.00            213.50            123.12   
NUM_CPF              426.00    26830502524.11    19135846183.67   
CHAVE_CONTRATO       426.00            213.50            123.12   
RENDA_PRESUMIDA      426.00          10387.81         111783.91   
CEP                  426.00        4635212.33        1978839.32   
NUMERO               425.00            595.29           4869.60   
CEP_A                426.00        4635212.33        1978839.32   
LATITUDE             389.00            -23.58              0.08   
LONGITUDE            389.00            -46.60              0.11   
DDD_CELULAR          275.00             16.50             17.78   
CELULAR              275.00      976736587.11       56323647.16   
DDD_CELULAR_2        170.00             18.26             19.58   
CELULAR_2            170.00      960391066.33      138162785.25   
COD_BANCO            308.00            189.76            175.78   
NUM_AGENCIA          308.00           2570.55           2347.82   
NUM_CONTA            308.00       51862517.85      469016468.99   
CNPJ_CREDOR          421.00 31873758302107.55 22283968587101.65   
VALOR_DIVIDA         426.00         212868.56        2202077.89   
STATUS_CONSENTIMENTO 426.00              1.00              0.00   
IDADE                421.00             45.74            107.73   

                                 min              25%               50%  \
Unnamed: 0                      1.00           107.25            213.50   
NUM_CPF                  94700508.00   10869067862.00    26041138816.00   
CHAVE_CONTRATO                  1.00           107.25            213.50   
RENDA_PRESUMIDA                 0.00          2068.50           2431.58   
CEP                       1007020.00       3153250.25        4323120.00   
NUMERO                          3.00            58.00            164.00   
CEP_A                     1007020.00       3153250.25        4323120.00   
LATITUDE                      -23.80           -23.63            -23.56   
LONGITUDE                     -46.81           -46.68            -46.63   
DDD_CELULAR                    11.00            11.00             11.00   
CELULAR                  78462120.00     969555131.50      982497429.00   
DDD_CELULAR_2                  11.00            11.00             11.00   
CELULAR_2                77151941.00     973504991.25      982267694.50   
COD_BANCO                       1.00            33.00            237.00   
NUM_AGENCIA                     1.00           563.25           1900.00   
NUM_CONTA                     290.00         15030.00          53951.00   
CNPJ_CREDOR          113451000135.00 8623036000133.00 40432544000147.00   
VALOR_DIVIDA                    0.00           198.50            360.00   
STATUS_CONSENTIMENTO            1.00             1.00              1.00   
IDADE                          17.00            30.00             37.00   

                                   75%               max  
Unnamed: 0                      319.75            426.00  
NUM_CPF                 35798933865.00    99921022504.00  
CHAVE_CONTRATO                  319.75            426.00  
RENDA_PRESUMIDA                2923.55        1890000.00  
CEP                         5644782.50        8490018.00  
NUMERO                          430.00          99999.00  
CEP_A                       5644782.50        8490018.00  
LATITUDE                        -23.51            -23.43  
LONGITUDE                       -46.52            -46.38  
DDD_CELULAR                      11.00             98.00  
CELULAR                   992424488.50      999993969.00  
DDD_CELULAR_2                    11.00             98.00  
CELULAR_2                 993495586.00      999889639.00  
COD_BANCO                       341.00            756.00  
NUM_AGENCIA                    3781.00           9270.00  
NUM_CONTA                    488489.25     6500059293.00  
CNPJ_CREDOR          43374768000138

In [93]:
#Preprocessing
# mudar o formato da variável number para string
cadastro['NUM_CPF'] = cadastro['NUM_CPF'].astype(str)
cadastro['CHAVE_CONTRATO'] = cadastro['CHAVE_CONTRATO'].astype(str)
cadastro['CEP'] = cadastro['CEP'].astype(str)
cadastro['CEP_A'] = cadastro['CEP_A'].astype(str)
cadastro['NUMERO'] = cadastro['NUMERO'].astype(str)
cadastro['LATITUDE'] = cadastro['LATITUDE'].astype(str)
cadastro['LONGITUDE'] = cadastro['LONGITUDE'].astype(str)
cadastro['CELULAR'] = cadastro['CELULAR'].astype(str)
cadastro['DDD_CELULAR'] = cadastro['DDD_CELULAR'].astype(str)
cadastro['CELULAR_2'] = cadastro['CELULAR_2'].astype(str)
cadastro['DDD_CELULAR_2'] = cadastro['DDD_CELULAR_2'].astype(str)
cadastro['COD_BANCO'] = cadastro['COD_BANCO'].astype(str)               
cadastro['NUM_AGENCIA'] = cadastro['NUM_AGENCIA'].astype(str)          
cadastro['NUM_CONTA'] = cadastro['NUM_CONTA'].astype(str)        
cadastro['CNPJ_CREDOR'] = cadastro['CNPJ_CREDOR'].astype(str)        
cadastro['STATUS_CONSENTIMENTO'] = cadastro['STATUS_CONSENTIMENTO'].astype(str)


In [94]:
# Tipos de variáveis
cadastro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 426 entries, 0 to 425
Data columns (total 30 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            426 non-null    int64  
 1   NUM_CPF               426 non-null    object 
 2   CHAVE_CONTRATO        426 non-null    object 
 3   DATA_NASCIMENTO       426 non-null    object 
 4   RENDA_PRESUMIDA       426 non-null    float64
 5   CEP                   426 non-null    object 
 6   LOGRADOURO            426 non-null    object 
 7   NUMERO                426 non-null    object 
 8   COMPLEMENTO           206 non-null    object 
 9   CEP_A                 426 non-null    object 
 10  BAIRRO                390 non-null    object 
 11  CIDADE                426 non-null    object 
 12  UF                    426 non-null    object 
 13  LATITUDE              426 non-null    object 
 14  LONGITUDE             426 non-null    object 
 15  DDD_CELULAR           4

In [95]:
# Análise descritiva das variáveis
cadastro.describe().round(2)

Unnamed: 0  RENDA_PRESUMIDA  VALOR_DIVIDA   IDADE
count      426.00           426.00        426.00  421.00
mean       213.50         10387.81     212868.56   45.74
std        123.12        111783.91    2202077.89  107.73
min          1.00             0.00          0.00   17.00
25%        107.25          2068.50        198.50   30.00
50%        213.50          2431.58        360.00   37.00
75%        319.75          2923.54        981.00   46.00
max        426.00       1890000.00   28072009.00 2006.00

In [96]:
# Soma o número de dados faltantes (missing value)
cadastro.isna().sum()

Unnamed: 0                0
NUM_CPF                   0
CHAVE_CONTRATO            0
DATA_NASCIMENTO           0
RENDA_PRESUMIDA           0
CEP                       0
LOGRADOURO                0
NUMERO                    0
COMPLEMENTO             220
CEP_A                     0
BAIRRO                   36
CIDADE                    0
UF                        0
LATITUDE                  0
LONGITUDE                 0
DDD_CELULAR               0
CELULAR                   0
DDD_CELULAR_2             0
CELULAR_2                 0
COD_BANCO                 0
NUM_AGENCIA               0
NUM_CONTA                 0
MARCA_VEICULO           424
MODELO_VEICULO          424
PLACA_VEICULO           424
CNPJ_CREDOR               0
VALOR_DIVIDA              0
STATUS_CONSENTIMENTO      0
DATA_INCLUSAO1            5
IDADE                     5
dtype: int64

In [97]:
# Transpor as duas primeiras linhas da base de dados para visalização
cadastro.head(2).T

0                 1
Unnamed: 0                                     1                 2
NUM_CPF                              40977545806       72544350725
CHAVE_CONTRATO                                 1                 2
DATA_NASCIMENTO                       25/09/1996        10/03/1961
RENDA_PRESUMIDA                       1890000.00        1333890.24
CEP                                      2471020           1319001
LOGRADOURO            R AUGUSTO VIANA DE CASTELO     R MARIA PAULA
NUMERO                                      26.0             201.0
COMPLEMENTO                                  NaN             AP 52
CEP_A                                    2471020           1319001
BAIRRO                             VILA BASILEIA        BELA VISTA
CIDADE                                 SAO PAULO         SAO PAULO
UF                                            SP                SP
LATITUDE                                     nan               nan
LONGITUDE                                    nan               nan
DDD_CELULAR                                 11.0              11.0
CELULAR                              977060299.0       972545525.0
DDD_CELULAR_2                                nan              11.0
CELULAR_2                                    nan       985714369.0
COD_BANCO                                    nan             341.0
NUM_AGENCIA                                  nan            6571.0
NUM_CONTA                                    nan           15044.0
MARCA_VEICULO                                NaN           FERRARI
MODELO_VEICULO                               NaN        355 SPIDER
PLACA_VEICULO                                NaN           AFZ6118
CNPJ_CREDOR                      3774819001508.0  40432544000147.0
VALOR_DIVIDA                                 220               850
STATUS_CONSENTIMENTO                           1                 1
DATA_INCLUSAO1                2/26/2019 11:28:49  2/7/2001 0:00:00
IDADE                                      23.00             40.00

In [98]:
# Mudar a categoria "nan" para "NaN" (identificação de missing data)
cadastro.replace('nan', np.NaN, inplace=True)

In [99]:
# Transpor as duas primeiras linhas da base de dados para visalização
cadastro.head(2).T

0                 1
Unnamed: 0                                     1                 2
NUM_CPF                              40977545806       72544350725
CHAVE_CONTRATO                                 1                 2
DATA_NASCIMENTO                       25/09/1996        10/03/1961
RENDA_PRESUMIDA                       1890000.00        1333890.24
CEP                                      2471020           1319001
LOGRADOURO            R AUGUSTO VIANA DE CASTELO     R MARIA PAULA
NUMERO                                      26.0             201.0
COMPLEMENTO                                  NaN             AP 52
CEP_A                                    2471020           1319001
BAIRRO                             VILA BASILEIA        BELA VISTA
CIDADE                                 SAO PAULO         SAO PAULO
UF                                            SP                SP
LATITUDE                                     NaN               NaN
LONGITUDE                                    NaN               NaN
DDD_CELULAR                                 11.0              11.0
CELULAR                              977060299.0       972545525.0
DDD_CELULAR_2                                NaN              11.0
CELULAR_2                                    NaN       985714369.0
COD_BANCO                                    NaN             341.0
NUM_AGENCIA                                  NaN            6571.0
NUM_CONTA                                    NaN           15044.0
MARCA_VEICULO                                NaN           FERRARI
MODELO_VEICULO                               NaN        355 SPIDER
PLACA_VEICULO                                NaN           AFZ6118
CNPJ_CREDOR                      3774819001508.0  40432544000147.0
VALOR_DIVIDA                                 220               850
STATUS_CONSENTIMENTO                           1                 1
DATA_INCLUSAO1                2/26/2019 11:28:49  2/7/2001 0:00:00
IDADE                                      23.00             40.00

In [100]:
# Soma o número de dados faltantes (missing value)
cadastro.isna().sum()

Unnamed: 0                0
NUM_CPF                   0
CHAVE_CONTRATO            0
DATA_NASCIMENTO           0
RENDA_PRESUMIDA           0
CEP                       0
LOGRADOURO                0
NUMERO                    1
COMPLEMENTO             220
CEP_A                     0
BAIRRO                   36
CIDADE                    0
UF                        0
LATITUDE                 37
LONGITUDE                37
DDD_CELULAR             151
CELULAR                 151
DDD_CELULAR_2           256
CELULAR_2               256
COD_BANCO               118
NUM_AGENCIA             118
NUM_CONTA               118
MARCA_VEICULO           424
MODELO_VEICULO          424
PLACA_VEICULO           424
CNPJ_CREDOR               5
VALOR_DIVIDA              0
STATUS_CONSENTIMENTO      0
DATA_INCLUSAO1            5
IDADE                     5
dtype: int64

In [101]:
# Calcular o porcentagem de dados faltantes (missing value)
cadastro.isna().sum()/len(cadastro)*100

Unnamed: 0              0.00
NUM_CPF                 0.00
CHAVE_CONTRATO          0.00
DATA_NASCIMENTO         0.00
RENDA_PRESUMIDA         0.00
CEP                     0.00
LOGRADOURO              0.00
NUMERO                  0.23
COMPLEMENTO            51.64
CEP_A                   0.00
BAIRRO                  8.45
CIDADE                  0.00
UF                      0.00
LATITUDE                8.69
LONGITUDE               8.69
DDD_CELULAR            35.45
CELULAR                35.45
DDD_CELULAR_2          60.09
CELULAR_2              60.09
COD_BANCO              27.70
NUM_AGENCIA            27.70
NUM_CONTA              27.70
MARCA_VEICULO          99.53
MODELO_VEICULO         99.53
PLACA_VEICULO          99.53
CNPJ_CREDOR             1.17
VALOR_DIVIDA            0.00
STATUS_CONSENTIMENTO    0.00
DATA_INCLUSAO1          1.17
IDADE                   1.17
dtype: float64

## Imputação de dados faltantes

In [103]:
cadastro["IDADE"].describe()

count    421.00
mean      45.74
std      107.73
min       17.00
25%       30.00
50%       37.00
75%       46.00
max     2006.00
Name: IDADE, dtype: float64

In [104]:
# Imputação de dados faltantes usando a média
cadastro['Idade_i'] = cadastro['IDADE'].fillna(value=cadastro['IDADE'].median())

In [105]:
# Análise descritiva das variáveis
cadastro.describe().round(2)

Unnamed: 0  RENDA_PRESUMIDA  VALOR_DIVIDA   IDADE  Idade_i
count      426.00           426.00        426.00  421.00   426.00
mean       213.50         10387.81     212868.56   45.74    45.63
std        123.12        111783.91    2202077.89  107.73   107.10
min          1.00             0.00          0.00   17.00    17.00
25%        107.25          2068.50        198.50   30.00    30.00
50%        213.50          2431.58        360.00   37.00    37.00
75%        319.75          2923.54        981.00   46.00    46.00
max        426.00       1890000.00   28072009.00 2006.00  2006.00

In [106]:
# Cálculo do coeficiente de variação (CV)
# CV = desvio padrão/média

desc = cadastro.describe().T

desc['CV'] = desc['std']/desc['mean']

In [107]:
desc.round(2)

count      mean        std   min     25%     50%     75%  \
Unnamed: 0      426.00    213.50     123.12  1.00  107.25  213.50  319.75   
RENDA_PRESUMIDA 426.00  10387.81  111783.91  0.00 2068.50 2431.58 2923.54   
VALOR_DIVIDA    426.00 212868.56 2202077.89  0.00  198.50  360.00  981.00   
IDADE           421.00     45.74     107.73 17.00   30.00   37.00   46.00   
Idade_i         426.00     45.63     107.10 17.00   30.00   37.00   46.00   

                        max    CV  
Unnamed: 0           426.00  0.58  
RENDA_PRESUMIDA  1890000.00 10.76  
VALOR_DIVIDA    28072009.00 10.34  
IDADE               2006.00  2.36  
Idade_i             2006.00  2.35

# Interpretação do CV

Renda presumida: O desvio padrão é 10.76 vezes o valor da média. Isso indica presença de outliers. Quantos outliers?


    Recomendado é até 0,3 ou 30%. Valores acima de 30% indicam presença de outliers. (Critério do IBGE)
    

## SWEETVIZ_REPORT

In [110]:
!pip install sweetviz

In [111]:
import sweetviz as sv

In [112]:
my_report = sv.analyze(cadastro)
my_report.show_html() # Default arguments will generate to "SWEETVIZ_REPORT.html"

                                             |      | [  0%]   00:00 -> (? left)

Report SWEETVIZ_REPORT.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.
